events-api/src/modules/events/events.controller.ts

```
@@ -47,4 +47,52 @@ export class EventsController {
       data: foundEvent,
     };
   }
+
+  @Patch(':eventId')
+  @Roles(UserRole.Admin)
+  async updateEventById(
+    @Param(
+      'eventId',
+      new ParseUUIDPipe({
+        errorHttpStatusCode: HttpStatus.BAD_REQUEST,
+        version: '4',
+      }),
+    )
+    eventId: string,
+    @Body() updateEventDto: Partial<CreateEventReqDto>,
+  ) {
+    const foundEvent = await this.eventsService.getEventById(eventId);
+
+    if (!foundEvent) {
```

Why not to throw an error on the service level?